src/compiler/checker.ts

```diff
@@ -27006,14 +27095,26 @@ namespace ts {
                 return restType;
             }
             const types = [];
-            const names = [];
+            let names: (NamedTupleMember | ParameterDeclaration)[] | undefined = [];
             for (let i = pos; i < nonRestCount; i++) {
                 types.push(getTypeAtPosition(source, i));
-                names.push(getParameterNameAtPosition(source, i));
+                const name = getNameableDeclarationAtPosition(source, i);

```

Comment:
Correct! That does mean that generated names can still be shoved into parameter lists (like `arg_0`) but will disappear when copied into a tuple. I actually think this is a good thing, since the generated names were just positional anyway.